src/Model/Casts/BackedEnumCast.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Mongolid\Model\Casts;
+
+use BackedEnum;
+use Mongolid\Model\Casts\Exceptions\InvalidTypeException;
+use Throwable;
+
+class BackedEnumCast implements CastInterface
+{
+    /**
+     * @param class-string<BackedEnum> $backedEnum
+     */
+    public function __construct(
+        private string $backedEnum
+    ) {
+    }
+
+    public function get(mixed $value): ?BackedEnum
+    {
+        if (is_null($value)) {
+            return null;
+        }
+
+        return ($this->backedEnum)::from($value);
+    }
+
+    public function set(mixed $value): string|int|null
```

Comment:
This method doesn't look like a set but a get

Reply:
Acho que isso tem que ser discutido lá no PR dos datetime casts, pois lá foi implementado o mecanismo de plugar novos casters, aqui só estou implementando a interface.

Reply:
Coloquei um comentário lá também.
https://github.com/leroy-merlin-br/mongolid/pull/220#discussion_r1365503134